scipy/fft/_pocketfft/pocketfft_hdronly.h

```diff
@@ -34,11 +34,15 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define POCKETFFT_HDRONLY_H
 
 #ifndef __cplusplus
-#error This file is C++ and requires a C++ compiler
+#error This file is C++ and requires a C++ compiler.
 #endif
 
 #if !(__cplusplus >= 201103L || _MSVC_LANG+0L >= 201103L)
-#error This file requires at least C++11 support
+#error This file requires at least C++11 support.
+#endif
+
+#ifndef POCKETFFT_CACHE_SIZE
+#define POCKETFFT_CACHE_SIZE 16
```

Comment:
16 is the number of plans to keep in the cache? That does not seem like very many to me, but is probably okay.

Reply:
I had a look and fftpack only caches up to 10 plans. Also note that this is per dtype.

Reply:
I chose a fairly small number since the plans are not exactly small; they contain the full twiddle factor array, which is as large as the input data for 1D FFTs (even larger than that when the Bluestein algorithm is used). So if someone runs different very long 1D FFTs, they might wonder where all their memory has gone...
It might be worth considering to cache only plans below a given maximum length (maybe up to 2**20?)...